### PR TITLE
[kubernetes] Add support for k8s snap

### PR DESCRIPTION
**Pull request by arif-ali**
Also, fix command for microk8s, and use variable

---
Please place an 'X' inside each '[]' to confirm you adhere to our [Contributor Guidelines](https://github.com/sosreport/sos/wiki/Contribution-Guidelines)

- [x] Is the commit message split over multiple lines and hard-wrapped at 72 characters?
- [x] Is the subject and message clear and concise?
- [x] Does the subject start with **[plugin_name]** if submitting a plugin patch or a **[section_name]** if part of the core sosreport code?
- [x] Does the commit contain a **Signed-off-by: First Lastname <email@example.com>**?
- [x] Are any related Issues or existing PRs [properly referenced](https://docs.github.com/en/issues/tracking-your-work-with-issues/creating-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) via a Closes (Issue) or Resolved (PR) line?
- [x] Are all passwords or private data gathered by this PR [obfuscated](https://github.com/sosreport/sos/wiki/How-to-Write-a-Plugin#how-to-prevent-collecting-passwords)?

### CODE REVIEW

**Review by jcastill — APPROVED** [review]

File: sos/report/plugins/k8s.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+        self.add_copy_spec([
+            "/etc/kubernetes/"
+            f"{k8s_common}/args",
+            f"{k8s_common}/var/lib/k8s-dqlite/info.yaml",
+            f"{k8s_common}/var/lib/k8s-dqlite/cluster.yaml",
+            f"{k8s_common}/var/lib/k8sd/state/truststore/k8s.yaml",
+            f"{k8s_common}/var/lib/k8sd/state/database/info.yaml",
+            f"{k8s_common}/var/lib/k8sd/state/database/cluster.yaml",
+            f"{k8s_common}/var/lib/k8sd/state/daemon.yaml",
```
Comment (pmoravec) [inline sos/report/plugins/k8s.py:48]: The `yaml` files cant contain any credentials worth to `postproc`, right?

Comment (arif-ali) [reply sos/report/plugins/k8s.py:48]: I have gone through these files maually twice :), and there isn't anything to obfuscate, except for `/var/snap/k8s/common/var/lib/k8sd/state/truststore/k8s.yaml`, which I've already added in `postproc()`

I am now getting all of this verified by our k8s SMEs to ensure we got everything

**Comment by TurboTurtle** [comment]
Plugin looks fine. My only concern is around naming. We have `kubernetes` as well as `microk8s` already today. `k8s` is a well-known shorthand for `kubernetes`, so it may start being a little opaque what plugin collects what for "$defacto_standard_container_orchestration_platform"

**Comment by arif-ali** [comment]
I've moved it now to the `kubernetes.py` plugin, hopefully that makes much more sense. Always good to think outside of the box, and this does make better sense.

This also made me realise that we were collecting a folder, and obfuscating the said folder. So, actually needed less code adding.

Maybe we can also move the `microk8s`  here, and have everything under the one umbrella, I'll leave that thought here, and see what others think

**Review by TurboTurtle — APPROVED** [review]

**Pull request merged**